arcs/src/systems/spatial_relation.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{
+    components::{SpatialEntity, DrawingObject, Space},
+    algorithms::{Bounded},
+};
+use specs::prelude::*;
+
+/// A [`System`] which keeps track of the spatial relation of entities
+#[derive(Debug)]
+pub struct SpatialRelation {
+    changes: ReaderId<ComponentEvent>,
+    to_insert: BitSet,
+    to_update: BitSet,
+}
+
+impl SpatialRelation {
+    pub const NAME: &'static str = module_path!();
+
+    pub fn new(world: &World) -> Self {        
+        SpatialRelation {
+            changes: world.write_storage::<DrawingObject>().register_reader(),
```

Comment:
If we monitor the `BoundingBox` storage instead of `DrawingObject` it should let us avoid re-calculating bounding boxes every time something gets moved.

That means our [`arcs::systems::register_background_tasks()`](https://github.com/Michael-F-Bryan/arcs/blob/5528734e2843308d3e207627d3267529493e4b9f/arcs/src/systems/mod.rs#L12) function will need to be updated so the `SpatialRelation` is added after `SyncBounds`, and we'll need to tell the dispatcher it depends on `SyncBounds::NAME` (that's what `with`'s final parameter is for).

Reply:
Done